// local-ssl-proxy/dist/index.js
'use strict';

function _interopDefault (ex) { return 'default' in ex ? ex['default'] : ex; }

var nomnom = _interopDefault(require('nomnom'));
var Path = _interopDefault(require('path'));
var fs = _interopDefault(require('fs'));

var name = "local-ssl-proxy";
var version = "1.2.3";
var author = "Cameron Hunter <hello@cameronhunter.co.uk>";
var description = "Simple SSL HTTP proxy using a self-signed certificate. Intended for local development only.";
var repository = { "type": "git", "url": "http://github.com/cameronhunter/local-ssl-proxy.git" };
var license = "MIT";
var bin = { "local-ssl-proxy": "bin/local-ssl-proxy" };
var main = "dist/index.js";
var scripts = { "build": "rollup -c", "pretest": "npm run build", "test": "ava" };
var dependencies = { "chalk": "^1.1.1", "http-proxy": "^1.13.1", "nomnom": "^1.8.1", "yargs": "^3.32.0" };
var devDependencies = { "ava": "^0.11.0", "babel-core": "^6.4.5", "babel-preset-es2015-rollup": "^1.1.1", "babel-preset-stage-1": "^6.3.13", "babel-register": "^6.4.3", "rollup": "^0.25.2", "rollup-plugin-babel": "^2.3.9", "rollup-plugin-json": "^2.0.0" };
var Package = {
	name: name,
	version: version,
	author: author,
	description: description,
	repository: repository,
	license: license,
	bin: bin,
	main: main,
	scripts: scripts,
	dependencies: dependencies,
	devDependencies: devDependencies
};

var exists = function exists(path) {
  return fs.accessSync(absolutePath(path));
};
var readFile = function readFile(path) {
  return fs.readFileSync(absolutePath(path), "utf8");
};
var absolutePath = function absolutePath(path) {
  return Path.isAbsolute(path) ? path : Path.resolve(process.cwd(), path);
};

var _commandline = nomnom.script(Package.name).options({
  hostname: {
    abbr: 'n',
    default: 'localhost'
  },
  source: {
    abbr: 's',
    default: 9001
  },
  target: {
    abbr: 't',
    default: 9000
  },
  version: {
    abbr: 'v',
    flag: true,
    transform: function transform() {
      return Package.version;
    }
  },
  cert: {
    abbr: 'c',
    default: Path.resolve(__dirname, '..', 'resources', 'localhost.cert'),
    callback: exists,
    transform: readFile
  },
  key: {
    abbr: 'k',
    default: Path.resolve(__dirname, '..', 'resources', 'localhost.key'),
    callback: exists,
    transform: readFile
  },
  config: {
    abbr: 'o',
    callback: exists,
    transform: readFile
  }
});

exports.commandline = _commandline;
